### PR TITLE
Chaineducation

### PR DESCRIPTION
**Pull request by muskanmarwah1**
#### Project Abstract

> ChainEducation is a decentralized platform where anyone can upgrade their web3 skills from hands on tutorials.
> They will be able to earn certificates as NFT once they complete a course.
> Platform will also create their profile to showcase it to the community. They can add their projects and achievements too.
> Educators can create their classroom. They can upload as many courses in a classroom.
> Students can get enrolled in a classroom. For every achievement, they will earn achievement NFT.
> Students can rate and review the courses too.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/grant_guidelines_per_category.md).
- [x] A BTC, Ethereum (USDT/USDC/DAI) or Polkadot/Kusama (aUSD) address for the payment of the milestones is provided inside the application.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the application. Here is some initial feedback: 

- Could you add the level of the application to the overview section, see the template https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md 
- Regarding the ink! Smart contracts, could you be more specific? What exactly do you need to develop? Why can’t you use the existing smart contracts, here for example https://github.com/paritytech/ink/tree/master/examples (erc20, erc721) 
- I can not open your figma link: https://www.figma.com/file/1YtoEah6gT56bgKKDw66bR/Untitled?node-id=0%3A1&fuid=867329998874333171 
- In general could you add more technical details to the deliveries (e.g. programming language, etc.)?

**Comment by muskanmarwah1** [comment]
Hey, thanks a lot for the initial feedback. I have updated the grant application. Requesting you to please go through it.

 - Added the level of application.
 - Yes, we will be using the smart contracts you mentioned, along with two custom smart contracts that we will develop. Please check the updated application.
 - I had a word with my designer. Right now the designs are not complete but feel free to check our website and our Job Board.
 Website link: https://chaineducation.xyz/
 Job Board link: https://chaineducation.niceboard.co/
- Added the programming language for smart contract, frontend, and backend. Please check. Let me know if more technical details are required.

**Comment by muskanmarwah1** [comment]
Also, we have started to push code to our repos actively. Kindly check our repos:
 - https://github.com/chain-education/smart-contracts
 - https://github.com/chain-education/frontend
 - https://github.com/chain-education/backend

**Comment by muskanmarwah1** [comment]
Hey @Noc2. I have added a few mockups in the application. Please check them.

**Comment by muskanmarwah1** [comment]
Hey @Noc2. Any update on this?

**Comment by muskanmarwah1** [comment]
Also, may I know your working days?
I wait for your reply daily :(

**Review by Noc2 — CHANGES_REQUESTED** [review]
Sorry for the delay. We currently have a big backlog of grant deliveries, grants applications as well as teams reaching out to us regarding multiple questions. Also part of the team is OOO at the moment. I have a few follow-up questions: 

- Regarding the two new smart contracts, why do you need to develop a smart contract for achievements NFTs? Why aren’t you using ERC-721 for this? And is it a good idea to store user information on-chain? You could also implement this as an NFT. 
- Regarding the UI: What kind of on-chain information will the front-end use? Could you maybe specify this as part of the deliveries? 

Otherwise, the application looks good to me.

**Comment by muskanmarwah1** [comment]
Hey @Noc2. I have updated the application also. Please review.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the super quick reply. I will mark the application as ready for review to speed up the evaluation process. 

But regarding your first delivery, it’s still unclear to me what kind of smart contract you would actually develop as part of the delivery (functions of it). Keep in mind, we mostly support open-source software development via this grant program, and not the reuse of existing smart contracts. So this is the most important part of the delivery. 
Regarding your second milestone, could you clarify as part of the delivery what “etc.” means and if this data is automatically shown based on-chain data or if it is something that manually needs to be added to the page?

**Comment by muskanmarwah1** [comment]
> Thanks for the super quick reply. I will mark the application as ready for review to speed up the evaluation process.

Thanks a lot. Really grateful.

> But regarding your first delivery, it’s still unclear to me what kind of smart contract you would actually develop as part of the delivery (functions of it). Keep in mind, we mostly support open-source software development via this grant program, and not the reuse of existing smart contracts. So this is the most important part of the delivery.

The smart contract source code will be 100% open source and assure you that existing smart contracts won't be reused.

This smart contract will be used to distribute erc-20 tokens to the users. Users will be able to pay using tokens if part of the course is not free. Earnings will directly go to the creator of the course.
Apart from that, it will also be used to mint NFT for the users who successfully achieve the given target.

> Regarding your second milestone, could you clarify as part of the delivery what “etc.” means and if this data is automatically shown based on-chain data or if it is something that manually needs to be added to the page?

Apart from achievement NFTs, certificates, profiles, and wallet information. it will also show the total exercises completed, the current streak of a user, and social handles.

Things that need to be manually added like:
 - Name
 - Past Experience
 - Skills

**Comment by avirajkhare00** [comment]
Hey @Noc2. Any update?

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for pinging me here as well. I only now realized that you are applying for two grants at the same time. We usually only allow grantees initially to work on one grant at the same time. Simply to get a better understanding of the team and later we can sign follow-up grants. For reference: 

- https://github.com/w3f/Grants-Program/pull/1201 
- https://github.com/w3f/Grants-Program/pull/1192 

Regarding your answer above: Distributing NFTs and tokens is something that in most cases is done via some off-chain software/tools. If you develop a smart contract for this, that automatically distributes these tokens or NFTs, you would also need to develop an oracle that provides data to this smart contract in a decentralized way. Otherwise, there is no benefit of having this contract in the first place imho.

**Review by semuelle — CHANGES_REQUESTED** [review]
To be blunt: there is so much wrong with this application that I consider it irrecoverable. You "assure [us] that existing smart contracts won't be reused", as if that's a good thing. You add an oracle to the list of deliverables without any thought how to accomplish that and what it's supposed to do. Up to that point, it was a milestone with ERC20 and ERC721 tokens for 10,000 USD. It is still unclear what it is you need to do, other than using the existing standards, because the details you added aren't technical. How are you going to publish tutorials in M1 if frontend and API are developed in M2 and M3? `API Design` and `Database Modelling` is in the last milestone. There is no word on existing similar platforms, why it should be on IPFS and how you are planning to win anyone over to create courses on your platform.

I will not be supporting this application.

File: applications/chaineducation.md
Comment (avirajkhare00) [reply applications/chaineducation.md]: After many discussions, we will use **DOT**, which is native to polkadot. We want to use DOT because if users buys a course, these tokens will directly go to the creator's account.

File: applications/chaineducation.md
Comment (avirajkhare00) [reply applications/chaineducation.md]: Apologies for the fact that we blindly copy-pasted it from the sample application. We have fixed documentation in every milestone.

**Comment by avirajkhare00** [comment]
> To be blunt: there is so much wrong with this application that I consider it past recovery. You "assure [us] that existing smart contracts won't be reused", as if that's a good thing. You add an oracle to the list of deliverables without any thought how to accomplish that and what it's supposed to do. Up to that point, it was a milestone with ERC20 and ERC721 tokens for 10,000 USD. It is still unclear what it is you need to do, other than using the existing standards, because the details you added aren't technical. How are you going to publish tutorials in M1 if frontend and API are developed in M2 and M3? API Design and Database Modelling is in the last milestone. There is no word on existing similar platforms, why it should be on IPFS and how you are planning to win anyone over to create courses on your platform.

You are correct. We wrote, "existing smart contract won't be reused." As asked by @Noc2, it is an important part of the delivery. You can check here: https://github.com/w3f/Grants-Program/pull/1201#pullrequestreview-1140939902

Also, the last milestone contains the creation of a smart contract responsible for minting the NFTs so that it can give NFTs to a user who completes certain milestones.

We need $10,000 for our first milestone because we need to get our custom smart contract audited, which will cost us $6,000-$8,000. Most of the money from the first milestone will go toward that as this smart contract is critical.

We will be publishing tutorials to use our smart contracts in the M1. In M2, we will publish the tutorial on testing, running, and deploying the frontend. In the M3, we will publish the tutorial on testing, running, and deploying the backend code.

Yes, API design and DB modeling are in the last part because it is part of the backend.

We have added both similar platforms and how our platform is unique in the application. Please check it.

Only the user profiles will be on IPFS. Our course videos will be served via Amazon S3 as the storage will be cheap. We have no intention of pushing the course files on IPFS. Then it will be publically available, and creators won't be able to earn money.

> I will not be supporting this application.

@semuelle. We respect your decision. May we know why and how we can improve?
I request you, please re-check the application once and provide us the feedback. The feedback you gave us is extremely useful, and we corrected the application based on your feedback.

**Comment by avirajkhare00** [comment]
@muskanmarwah1 has updated the DB schema and API spec in the application. Kindly check

File: applications/chaineducation.md
Comment (avirajkhare00) [reply applications/chaineducation.md]: This is outdated. We have updated the application. We don't need oracle anywhere.

File: applications/chaineducation.md
Comment (semuelle) [inline applications/chaineducation.md:165]: This is a list of deliverables. What is being delivered here? Also, which chain are you planning to deploy on? Polkadot doesn't support smart contracts.

Comment (avirajkhare00) [reply applications/chaineducation.md:165]: You are correct here. Polkadot doesn't support smart contracts. We will be using parachain to develop it. Then we will use relay chain to connect parachain with the main polkadot network.

Comment (avirajkhare00) [reply applications/chaineducation.md:165]: We will be deploying these above two smart contracts. Third smart contract needs to be coded from the scratch. @muskanmarwah1 is updating the application.

**Comment by avirajkhare00** [comment]
> Your application, edits and comments suggest that there is a lack of knowledge of the Substrate architecture and the work it requires to build a dapp and a parachain. On top of that, the architecture (user data on IPFS, centralised backend, NFTs on a parachain) and the changes you made to it during review seem random and the advantages it brings compared to existing platforms are slim in comparison to the disadvantages.

We are coming from the Ethereum ecosystem and yes, there is a lack of knowledge within us but learning on daily basis.
Give us 2-3 days, we will completely remove the need for centralization. It will be a complete dApp, not the hybrid one.

> I would very much support an educational platform that offers courses on Substrate and NFT credentials, but it would have to be decentralised from the ground up, have an existing user base or, at the very least, provide convincing arguments why it would need its own parachain when it would just as well work on any existing parachain using contracts.

We initially added milestones to create educational content once our dApp is up. We will focus on the Substrate ecosystem, but @Noc2 suggested we not include them since they were not the technical milestones. If you want, we can add them again.
We didn't specify anywhere that we will be creating our own parachain, not even in the milestones. We will be using the existing parachain.

**Comment by avirajkhare00** [comment]
> ...during review seem random and the advantages it brings compared to existing platforms are slim in comparison to the disadvantages.

It will take a few iterations for the disadvantages to go away.

**Comment by avirajkhare00** [comment]
> I would very much support an educational platform that offers courses on Substrate and NFT credentials, but it would have to be decentralised from the ground up

@semuelle Give us 2-3 days, and we will have a plan to execute this.

**Comment by avirajkhare00** [comment]
Hey, @Noc2 @semuelle, We have updated the application. Please provide us with feedback.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks again for the update and the reduced price. However, we usually don’t support the development of solidity smart contracts via our grants program. On the one hand, we are no experts when it comes to solidity and we don’t know what already exists etc. On the other hand, other grant programs, like [moonbeam](https://moonbeam.foundation/grants/), are a much better fit for these projects in our ecosystem. So my suggestion would be to apply for a grant from the moonbeam foundation in this case or focus on ink! or substrate instead.

**Comment by muskanmarwah1** [comment]
Sure. We will get back on Monday. We need to see all the pros and cons regarding this.

**Comment by avirajkhare00** [comment]
Dear @Noc2 
Thankyou for motivating us and for your constant support by your feedbacks. We would like to continue with W3F grant programme since we don't want Multichain capabilities of moonbeam. We would like to go ahead with W3F and will develop all the smart contracts using ink. We have updated our application a little for better understanding, and continuously working on the same thing. Really looking forward to work with W3F where our skills would be valued. Thankyou.

**Comment by avirajkhare00** [comment]
Hey @Noc2. Any update?

**Review by Noc2 — CHANGES_REQUESTED** [review]
Sorry for the delay. I started to write something here last Friday but forgot to post it, because of another meeting. 

- Could you remove this part: “We will be deploying our smart contracts on Moonbeam parachain.” We generally don’t want to fund the deployment of a contract on a specific parachain and not sure you still want to do this, given that you are now developing ink! Smart contracts. 
- It currently sounds like you are creating a lot of different smart contracts with your first delivery, but to me, it sounds more reasonable to have a single contract or two that supports this functionality.

**Comment by muskanmarwah1** [comment]
Hey @Noc2. Sorry for bothering you again and again. We removed the need to use Moonbeam network. We forgot to remove it. We will be using only two smart contracts. One is NFT, and the other one will contain the entire business logic.

File: applications/chaineducation.md
Comment (semuelle) [inline applications/chaineducation.md:193]: > A course will have multiple reviews.

Do only participants or finishers get to review? If so, wouldn't you need the registration/completion info also somewhere on-chain?

Comment (muskanmarwah1) [reply applications/chaineducation.md:193]: Students who do finish the entire course get to review the course. We will be doing this to maintain the quality of the reviews, and we do need unbiased reviews. Yes. The wallet address of the user will be the user_id. Also, we will be storing additional information about the user, like education, current location, if a user is open to working, and skills. The page will look somewhat like the CV of a user.

Comment (muskanmarwah1) [reply applications/chaineducation.md:193]: @semuelle Should I include all the data structures we will be using in the application?

Comment (semuelle) [reply applications/chaineducation.md:193]: If you have them, please do share. But often, data structures change a lot during the development phase, so we don't ask for it. I am asking, because it's more useful if the review process was decentralised, i.e. the eligible reviewers' addresses on-chain. 

How does the NFT distribution contract work? What triggers it? How do you prove to the smart contract that a person has completed the course?

Comment (muskanmarwah1) [reply applications/chaineducation.md:193]: > How does the NFT distribution contract work? What triggers it? How do you prove to the smart contract that a person has completed the course?

You asked a really nice question @semuelle. We will take a short quiz. If a person has completed a course, they should be able to answer 60% of the questions correctly. Also, the main smart contract will contain logic to execute Cross-Contract calling. It will mint new NFT.

> If you have them, please do share. But often, data structures change a lot during the development phase, so we don't ask for it. I am asking, because it's more useful if the review process was decentralised, i.e. the eligible reviewers' addresses on-chain.

The course review process is decentralized. I am sharing User structure below. Since change in struct is ever changing, I am including the struct here:
```
struct User {
    address: String,
    name: String,
    avatar_hash: String,
    courses_completed: u8,
    skills_id: Vec<u8>,
    location_id: u8,
    open_to_work: bool
}
```

Comment (semuelle) [reply applications/chaineducation.md:193]: > You asked a really nice question @semuelle. We will take a short quiz. If a person has completed a course, they should be able to answer 60% of the questions correctly.

- That means that there must be some action happening in the backend, right? Something/Someone signals the main smart contract that user was successful? Or how do you prevent others from minting an NFT? 
- What happens if the user fails? Can they redo the quiz as many times as they like?

Comment (avirajkhare00) [reply applications/chaineducation.md:193]: > That means that there must be some action happening in the backend, right? Something/Someone signals the main smart contract that user was successful? Or how do you prevent others from minting an NFT?

Frontend JavaScript code will trigger the NFTDeployer smart contract. We won't be using backend anywhere. Only smart contract owner(we) will be able to mint the NFT.

> What happens if the user fails? Can they redo the quiz as many times as they like?

Users can try as many times as they want.

Comment (avirajkhare00) [reply applications/chaineducation.md:193]: > That means that there must be some action happening in the backend, right? Something/Someone signals the main smart contract that user was successful? Or how do you prevent others from minting an NFT?

Frontend JavaScript code will trigger the NFTDeployer smart contract. We won't be using backend anywhere. Only smart contract owner(we) will be able to mint the NFT.

> What happens if the user fails? Can they redo the quiz as many times as they like?

Users can try as many times as they want.

Comment (avirajkhare00) [reply applications/chaineducation.md:193]: I want to point out that we will be manually minting NFT for people who will be completing this course for some time. Eventually, we will automate this process too. The smart contract will have the addresses of people who have completed the course.

**Comment by avirajkhare00** [comment]
@semuelle @Noc2 Any update sir?

**Comment by muskanmarwah1** [comment]
@Noc2, can we please have an update on this?
Thanks

**Comment by dsm-w3f** [comment]
@avirajkhare00 and @muskanmarwah1 thank you for your application. My apologies for the time to response, we have a quite long backlog of applications and deliveries to evaluate. We usually take some days to respond. 

Reading the application, I have some doubts about the requirements. Are they complete for an educational system?  When I read the API calls and features seems that you have functions to create and list things but not to edit and delete. Is this intentional? Or maybe there are some missing requirements? If they are missing, feel free to add new milestones and deliverables to the grant proposal, and budget to do it.

My colleagues already discussed many technical questions. I have just one, will the NFT for the diploma be a soul-bounded token? Usually, diplomas cannot be transferred more than once.

There are some learn-to-earn educational projects when the course creator pays the students to learn things. Today we have much free content on the internet, such as EdX, Coursera, courses on YouTube, etc. I don't know if content creators and students will be attracted to do courses on your platform just to pay with tokens and receive NFT at the end of the course. This looks like more a technological innovation than a business one. Maybe you could spend some time searching and digging into how to design this platform to be more business attractive. You can for example charge and give NFTs per content and not per course or another different thing. 

I'll be waiting for your answer. Sorry for saying that, be patient, It could take some days for me to answer back to you.

**Comment by muskanmarwah1** [comment]
> @avirajkhare00 and @muskanmarwah1 thank you for your application. My apologies for the time to response, we have a quite long backlog of applications and deliveries to evaluate. We usually take some days to respond.

Understood. No issues.

> Reading the application, I have some doubts about the requirements. Are they complete for an educational system? When I read the API calls and features seems that you have functions to create and list things but not to edit and delete. Is this intentional? Or maybe there are some missing requirements? If they are missing, feel free to add new milestones and deliverables to the grant proposal, and budget to do it.

This is a decentralized app from the ground up. We are not using backend anywhere hence deleted the APIs. This data will be stored inside IPFS in the form of JSON. Removing the file from IPFS is technically possible, but not all the IPFS nodes respect the file to be deleted, so it's it won't be possible to remove the file once it is created.

> My colleagues already discussed many technical questions. I have just one, will the NFT for the diploma be a soul-bounded token? Usually, diplomas cannot be transferred more than once.

Yes. We will give Soulbound NFT to the users who complete the course. We have updated this in the application. Thanks for pointing it out.

> There are some learn-to-earn educational projects when the course creator pays the students to learn things. Today we have much free content on the internet, such as EdX, Coursera, courses on YouTube, etc. I don't know if content creators and students will be attracted to do courses on your platform just to pay with tokens and receive NFT at the end of the course. This looks like more a technological innovation than a business one. Maybe you could spend some time searching and digging into how to design this platform to be more business attractive. You can for example charge and give NFTs per content and not per course or another different thing.

In the future, we will create DAO for chaineducation. We will put this feature for review in front of the DAO members. If they approve, we will have this functionality. For example, a company like Alchemy might not incentivize students with money. Rather, they will give a voucher or discount coupon. There are multiple ways to incentivize a user which we need to work upon.

> I'll be waiting for your answer. Sorry for saying that, be patient, It could take some days for me to answer back to you.

Thanks @dsm-w3f  for your review and response to this application. We will be waiting for your revert on this.

**Comment by avirajkhare00** [comment]
@muskanmarwah1 has added components of ChainEducation for better understanding. Please review them.
@semuelle @dsm-w3f @Noc2

**Comment by avirajkhare00** [comment]
@Noc2 any update on this?

**Review by Noc2 — APPROVED** [review]
Sorry again for the delay here. We are currently also on a company retreat. If you develop a soulbound nft in ink! you might also want to take a look at the following pallet, since it supports soulbound tokens:  https://github.com/rmrk-team/rmrk-substrate/blob/1b64a158d3a304bf96660fa8c29b035999329313/pallets/rmrk-core/src/lib.rs#L396 But I’m generally happy to go ahead with it.

**Comment by avirajkhare00** [comment]
Thanks a lot for approving our application. Sure, we will take a look at pallet you provided. Thanks for pointing it out.

**Comment by semuelle** [comment]
- You want to offer paid courses, but you don't explain how that would be accomplished in a decentralised way. 
- Your "ecosystem arms users with the knowledge to create dApps and Blockchains on top of Substrate ecosystem" and your target audience is people who want to use ink! or Substrate, yet you aren't building any courses yourself. 
- You want to offer "complete in-depth tutorials", yet don't mention anything about onboarding or incentivizing educators. 
- You say it's "a decentralized app from the ground up", but don't explain how you are going to implement signup, login, or paid courses. 
- You want to use soulbound tokens, but [not implement them yourselves](https://github.com/w3f/Grants-Program/pull/1201/files#diff-4d19006abbc0a4ec32d5a719c413f45220cbec2c96627a1e6492d882a10697f2R191).

I will stick with my [initial opinion](https://github.com/w3f/Grants-Program/pull/1201#pullrequestreview-1144828666) and not support this application.

**Review by semuelle — COMMENTED** [review]
- You want to offer paid courses, but you don't explain how that would be accomplished in a decentralised way. 
- Your "ecosystem arms users with the knowledge to create dApps and Blockchains on top of Substrate ecosystem" and your target audience is people who want to use ink! or Substrate, yet you aren't building any courses yourself. 
- You want to offer "complete in-depth tutorials", yet don't mention anything about onboarding or incentivizing educators. 
- You say it's "a decentralized app from the ground up", but don't explain how you are going to implement signup, login, or paid courses. 
- You want to use soulbound tokens, but [not implement them yourselves](https://github.com/w3f/Grants-Program/pull/1201/files#diff-4d19006abbc0a4ec32d5a719c413f45220cbec2c96627a1e6492d882a10697f2R191).

I will stick with my [initial opinion](https://github.com/w3f/Grants-Program/pull/1201#pullrequestreview-1144828666) and not support this application.

**Comment by avirajkhare00** [comment]
> You want to offer paid courses, but you don't explain how that would be accomplished in a decentralised way.

Users can send tokens to the smart contract, which will pay the course creator. This is how this functionality is going to work.

> Your "ecosystem arms users with the knowledge to create dApps and Blockchains on top of Substrate ecosystem" and your target audience is people who want to use ink! or Substrate, yet you aren't building any courses yourself.

Initially, we will create courses for building smart contracts using Ink and developing blockchains on the substrate ecosystem.

> You want to offer "complete in-depth tutorials", yet don't mention anything about onboarding or incentivizing educators.
You say it's "a decentralized app from the ground up", but don't explain how you will implement signup, login, or paid courses.

Our third team member, Yojit, working on the business side, is already in talks with the instructors. This issue will be sorted. We won't be covering anything regarding onboarding and incentivizing educators since we applied for this grant to develop the platform. We can include the business development part on this application if you want.

> You want to use soulbound tokens, but [not implement them yourselves](https://github.com/w3f/Grants-Program/pull/1201/files#diff-4d19006abbc0a4ec32d5a719c413f45220cbec2c96627a1e6492d882a10697f2R191).

That's the power of Open Source. If anything is already there, then why not leverage it?

> I will stick with my https://github.com/w3f/Grants-Program/pull/1201#pullrequestreview-1144828666 and not support this application.

We respect your decision. We need to understand what are the red flags in our project so that we can rectify them.

**Comment by avirajkhare00** [comment]
@semuelle let us know your thoughts.

**Comment by avirajkhare00** [comment]
Hey @Noc2. What are the next steps here?

**Comment by davidzindex999** [comment]
Hey @laboon. Can you please help us here?

**Comment by avirajkhare00** [comment]
Hey @laboon. Can you please help us here?

**Comment by Noc2** [comment]
Sorry for the late reply here. I’m currently OOO sick. But essentially you are/were waiting for additional people to approve your application.

**Comment by avirajkhare00** [comment]
Really sorry to hear about your health @Noc2  We were actually waiting for the approvals and worked really hard on this project. What do you suggest should we reopen the application ?

**Comment by Noc2** [comment]
It depends on you. You could reopen it and I share it again with the rest of the team. Maybe someone else is willing to approve it. We usually let applications open for 14 days before we close them

**Pull request closed without merging**
